libdevcrypto/Common.cpp

```diff
@@ -94,6 +110,54 @@ bool dev::decryptSym(Secret const& _k, bytesConstRef _cipher, bytes& o_plain)
 	return decrypt(_k, _cipher, o_plain);
 }
 
+h128 dev::encryptSymNoAuth(Secret const& _k, bytesConstRef _plain, bytes& o_cipher)
+{
+	h128 iv(Nonce::get());
+	return encryptSymNoAuth(_k, _plain, o_cipher, iv);
+}
+
+h128 dev::encryptSymNoAuth(Secret const& _k, bytesConstRef _plain, bytes& o_cipher, h128 const& _iv)
+{
+	o_cipher.resize(_plain.size());
```

Comment:
Not for AES. And it's also possible that _plain.data() == o_cipher.data().
The IV is external because it's only communicated for the first block. This is why both methods return the IV -- in CTR the IV is the counter and is incremented for every block. With ECIES we include the IV in the stream but for the framing protocol we don't because the IV is assumed on both sides to be zero.